contributors/devel/sig-instrumentation/logging.md

```diff
@@ -1,37 +1,73 @@
-## Logging Conventions
+## Logging
 
-The following conventions for the klog levels to use.
-[klog](http://godoc.org/github.com/kubernetes/klog) is globally preferred to
-[log](http://golang.org/pkg/log/) for better runtime control.
+This document will provide an overview on how Kubernetes community recommendation on logging. 
+Following recommendation should be followed in https://github.com/kubernetes/kubernetes, however any Kubernetes or external project is fill to follow guidelines below as most of the ideas can be applied more broadly.
 
-Shared libraries, such as `client-go`, should not use `klog.ErrorS()` and `klog.InfoS()`,
+### Logging in Kubernetes
+
+Kubernetes project uses [klog](https://github.com/kubernetes/klog) for logging.
+This library was created as a permanent fork of glog, as the original project was no longer developed. 
+However, should not be used by external projects as it is mostly developed to maintain backward compatibility.
+With introduction of [Structured Logging](https://github.com/kubernetes/enhancements/issues/1602) Kubernetes is being migrated to [logr](https://github.com/go-logr/logr) interface, any changes in klog are done to ensure smooth transition.
+
+For projects that want to integrate with Kubernetes or write same log format we recommend using [klogr](https://github.com/kubernetes/klog/tree/main/klogr), reimplementation of klog format that exposes [logr](https://github.com/go-logr/logr) interface.
+
+### Logging Conventions
+
+### When not to log?
+
+Shared libraries, such as `client-go`, should not log write logs themselves,
```

Comment:
```suggestion
Shared libraries, such as [`client-go`](https://github.com/kubernetes/client-go), should _not_
write logs themselves,
```

Reply:
A language question: what does "logs" stand for here and in other places ("Logs flags" in kube-apiserver)?

Is it short for "log files" or "log messages"?

"to log" is a verb, but here it is uses as a noun. I would use "logging" when I need a noun for the activity or "log messages" when I need an object ("write log messages").

Reply:
Historically the log was a noun. Originally it referred to a flotation device attached to a ship's waterproofed record book, but it has since come to mean the journal itself, or more commonly any broadly equivalent set of chronological, time-stampable records.
Further language drift added the verb “to log”. That use is fine too but it is actually a subsidiary meaning.

Reply:
>it has since come to mean the journal itself, or more commonly any broadly equivalent set of chronological, time-stampable records

So there is *one* of those. Does it then make sense to use the plural form or is singular better?

I understand "log books" (nautical), "log files" (computer science), but "logs" (plural for an abstract concept in computer science?) still sounds odd to me.

Reply:
Then I prefer "log messages" instead of the ambiguous "logs" abbreviation. I'm using "messages" instead of "entries" because "message" seems more common in computer science.

Reply:
I always interpreted "log message" as part of whole log that was provided by developer, so a log minus metadata like timestamp, caller etc.

Maybe we should also write down naming conventions for logging that all documentation should follow.